crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Lock
+    attr_accessor :logger
+    attr_accessor :path
+    attr_accessor :remote
+    attr_accessor :locked
+    attr_accessor :local_file
+
+    def initialize(options = {})
+      @logger = options.fetch :logger, Rails.logger
+      @path = options.fetch :path, Rails.root.join("tmp", "default.lock") # full path to lockfile
+      @remote = options.fetch :remote, false # or nodes array
+      @locked = false
+      @local_file = nil
+    end
+
+    def with_lock
+      acquire
+      yield if block_given?
+    ensure
+      release
+    end
+
+    def acquire
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for acquire is too high. [37.55/30]